### PR TITLE
[UI] Sessions Filter Logic and Session Details Screen

### PR DESCRIPTION
**Pull request by gissilali**
Worked on sessions screen filter logic and session details screen
does in a short paragraph._ please check the below boxes
- [x] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [x] I have tested the changes added on a physical device
- [x] I have run `./codeAnalysis.sh` to make sure all lint/formatting checks have been done.

View a short video of the feature here
https://photos.app.goo.gl/ZsBtjfwcArG56a717
Please add a screenshot (if necessary)

### CODE REVIEW

**Comment by wangerekaharun** [comment]
@gissilali resolve conflicts

File: data/build.gradle.kts [hunk 1]
```diff
@@ -70,9 +70,11 @@ dependencies {
     implementation(libs.room.runtime)
     ksp(libs.room.compiler)
     implementation(libs.room.ktx)
+    implementation(libs.kotlin.coroutines.android)
     implementation(libs.ktor.core)
     implementation(libs.ktor.android)
     implementation(libs.ktor.content.negotiation)
+    implementation(libs.gson.gson)
```
Comment (chepsi) [inline data/build.gradle.kts:77]: We don't neeed this gson lib

File: data/src/main/java/com/android254/data/network/Constants.kt [hunk 2]
```diff
@@ -21,5 +21,5 @@ object Constants {
     const val EVENT_SLUG = "droidconke-2022-281"
     const val ORG_SLUG = "droidcon-ke-645"
     const val EVENT_BASE_URL = "$BASE_URL/events/$EVENT_SLUG"
-    const val API_KEY = "droidconke-2020"
+    const val API_KEY = "droidconKe-2020"
```
Comment (chepsi) [inline data/src/main/java/com/android254/data/network/Constants.kt:24]: This one is hidden in the api_key file

File: data/src/main/java/com/android254/data/network/models/responses/Organizer.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    val twitter: String?,
+    val instagram: String?,
+    val logo: String?,
+    val slug: String?,
+    val status: String?,
+    val createdAt: String?,
+    val creator: Creator?,
+    val upcomingEventsCount: Int?,
+    val totalEventsCount: Int?
+)
```
Comment (chepsi) [inline data/src/main/java/com/android254/data/network/models/responses/Organizer.kt:20]: I think this model already exists.

File: presentation/src/main/java/com/android254/presentation/sessions/components/EventDaySelector.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun EventDaySelector(
+//    selected: String = droidconEventDays.first().value.format(
+//        DateTimeFormatter.ofPattern(
+//            "dd-MM-yyyy"
+//        )
+//    ),
+//    onSelect: (LocalDate) -> Unit = {},
```
Comment (chepsi) [inline presentation/src/main/java/com/android254/presentation/sessions/components/EventDaySelector.kt:47]: Delete?

**Review by chepsi — APPROVED** [review]

File: data/build.gradle.kts (hunk 1 above)
Comment (gissilali) [reply data/build.gradle.kts:77]: Will remove it.

File: data/src/main/java/com/android254/data/network/apis/SessionsApi.kt [hunk 5]
```diff
@@ -16,21 +16,30 @@
...
 
 class SessionsApi @Inject constructor(
-    private val client: HttpClient
+    private val client: HttpClient,
 ) {
-    suspend fun fetchSessions(perPage: Int = 200): PaginatedResponse<List<SessionDTO>> {
-        return client.get("${Constants.BASE_URL}/events/${Constants.EVENT_SLUG}/sessions") {
+    suspend fun fetchSessions(): EventScheduleResponse<Map<String, List<SessionApiModel>>> {
+        return client.get("${Constants.BASE_URL}/events/${Constants.EVENT_SLUG}/schedule") {
+            header("Api-Authorization-Key", "droidconKe-2020")
```
Comment (michaelbukachi) [inline data/src/main/java/com/android254/data/network/apis/SessionsApi.kt:32]: This header is not needed.

File: data/src/main/java/com/android254/data/network/apis/SessionsApi.kt [hunk 6]
```diff
@@ -16,21 +16,30 @@
...
-                parameters.append("per_page", perPage.toString())
+                parameters.append("grouped", "true")
             }
         }.body()
     }
+
+    suspend fun updateBookmarkedStatus(sessionId: String): BookmarkResponse {
+        return client.post("${Constants.BASE_URL}/events/${Constants.EVENT_SLUG}/bookmark_schedule/$sessionId") {
+            header("Api-Authorization-Key", "droidconKe-2020")
+            header("Authorization", "Bearer token")
```
Comment (michaelbukachi) [inline data/src/main/java/com/android254/data/network/apis/SessionsApi.kt:42]: These headers are not needed.

**Review by michaelbukachi — APPROVED** [review]

**Pull request merged**